### PR TITLE
normalize path  on android.

### PR DESCRIPTION
**Pull request by litefeel**
because AAssetManager_open do not support '../' in path.
for example:

can not open `test/../img.png` in android.
open `img.png` is ok.

this patch will change `test/../img.png` to `img.png`.

### CODE REVIEW

**Comment by minggo** [comment]
Change `test/../img.png` to `img.png` is strange.

**Comment by litefeel** [comment]
@minggo this patch is remove `..` in path.
for example:
old: `test/dir/../xx.png`
now: `test/xx.png`

old: `test/.../xx.png`
now: `xx.png`

old: `test/dir/dir/../../xx.png`
now:`test/xx.png`

exception:
old: `../xx.png`
now: `../xx.png`

old: `../test/../xx.png`
now: `../xx.png`

**Comment by minggo** [comment]
Yep, i know what is PR do.
I just curious  why should do it like this?
Developer should pass correct path, or it should get error information.
It is not a good idea  to translate it silently internal.

**Comment by joewan** [comment]
we have also met this problem,
I think we need to solve this bug,
or provide functions to normalize the filepath.

First,  Engine need to provide utils check whether the filepath is correct or not!
Second, Engine need to provide platform consistence for game developers, some platfoms support "..",  but some not. 

Just my own opinion above.

------------------ 原始邮件 ------------------
发件人: "minggo";notifications@github.com;
发送时间: 2014年9月30日(星期二) 中午11:09
收件人: "cocos2d/cocos2d-x"cocos2d-x@noreply.github.com; 

主题: Re: [cocos2d-x] remove '../' at path on android. (#8204)

Yep, i know what is PR do.
 I just curious  why should do it like this?
 Developer should pass correct path, or it should get error information.
 It is not a good idea  to translate it silently internal.

**Comment by litefeel** [comment]
@minggo 
i use cocostudio ui, for share plist, i put it to common dir.
and correct path to common dir in  .ExportJson.
so the path has `../`.

**Comment by minggo** [comment]
@joewan 
Yep, you are right.
I am considering offer a new file system that will handle common file operation.
Now `FileUtils` is simple.

**Comment by litefeel** [comment]
win and ios support `../`.
android not support `../`

**Comment by minggo** [comment]
Yep, Android is special. Its resources are in .apk after installing.

**Comment by minggo** [comment]
I have created a ticket of new file system here: http://discuss.cocos2d-x.org/t/about-file-system/17357.

File: cocos/platform/android/CCFileUtils-android.h [hunk 1]
```diff
@@ -61,6 +61,8 @@ class CC_DLL FileUtilsAndroid : public FileUtils
     /* override funtions */
     bool init();
 
+    std::string getNewFilename(const std::string &filename) const override;
```
Comment (WenhaiLin) [inline cocos/platform/android/CCFileUtils-android.h]: please add 'virtual' keyword

**Comment by litefeel** [comment]
@Dhilan007  added it.

**Comment by minggo** [comment]
Now there are two PRs for this issue.
I think we should close one.

@lite3 
@Dhilan007 
Any idea?

**Comment by litefeel** [comment]
@minggo @Dhilan007 
i think will close @Dhilan007 .

because the path can not cache when normalize path on `AAssetManager_open`.

**Comment by litefeel** [comment]
[ci rebuild]

**Comment by litefeel** [comment]
[ci rebuild]

**Comment by minggo** [comment]
> because the path can not cache when normalize path on AAssetManager_open

What did you mean?
Could you describe it in more detail?

**Comment by minggo** [comment]
Ok, i got your meaning.
But there is a problem of this PR.
Because getNewFileName is not used for this purpose, it is used to get a alias of a file name.

**Comment by litefeel** [comment]
@minggo 
i have a path `script.lua`, the real path is `script/script.lua`.
first times, search path to `script/script.lua`, and put _fullPathCache.
second times, get path from _fullPathCache.

**Comment by litefeel** [comment]
@minggo yes.
getNewFileName 
1. get a alias of a file name
2. normalize path

i think is ok.

**Comment by minggo** [comment]
Yep, you are right.

@Dhilan007 
Could you please test it?
And i will close your pull request.

**Comment by WenhaiLin** [comment]
@minggo It's work as expected.We can merge this pr.
@lite3 Thanks very much.

**Pull request merged**